src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -376,12 +412,11 @@ impl StaticHeader {
             (Err(_), Ok(buf_loc_2)) => {
                 // Copy 2 read OK, 1 resulted in IO Error
                 match StaticHeader::sigblock_from_buf(&buf_loc_2) {
-                    Ok(loc_2) => {
-                        if loc_2.is_some() {
-                            BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
-                        }
-                        Ok(loc_2)
-                    }
+                    Ok(loc_2) => Ok(if let Some(loc_2) = loc_2 {
+                        bda_write_check(f, &buf_loc_2, MetadataLocation::First, loc_2)
```

Comment:
There was a compilation failure here: 
```
error[E0308]: mismatched types
   --> src/engine/strat_engine/backstore/metadata/bda.rs:416:25
    |
416 |                         bda_write_check(f, &buf_loc_2, MetadataLocation::First, loc_2)
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `std::option::Option`, found enum `std::result::Result`
    |
    = note: expected type `std::option::Option<_>`
               found type `std::result::Result<std::option::Option<_>, stratis::errors::StratisError>`
```

Reply:
I didn't get one so I guess it was on a previous version?